backend/src/handlers/BaseHandler.ts

```diff
@@ -23,18 +25,43 @@ export abstract class BaseHandler<TPayload, TOutput extends BaseOutput> implemen
   /**
    *  @param validationSchema - The Joi schema that the payload will be validated against
    */
-  constructor(private validationSchema: Joi.ObjectSchema<TPayload>) {}
+  constructor(private validationSchema: Joi.ObjectSchema<TPayload>, private enumType: { [key: string]: string }) {}
 
   protected abstract getResult(payload: Result<TPayload>, res: Response<TOutput>): any;
 
   public runHandler(req: Request<TPayload>, res: Response<BaseOutput>) {
     const payload = validatePayload<TPayload>(this.validationSchema, req.body);
 
     if (payload.isError()) {
-      res.status(400).send({ ok: false, error: payload.getError() });
+      const error = payload.getError();
+      const apiError = mapBaseErrorToAPIError(error);
+
+      res.status(400).send({ ok: false, error: apiError });
       return;
     }
 
     return this.getResult(payload, res);
   }
+
+  protected handleError(error: IBaseError, res: Response) {
+    return ErrorHandler.handleError(error, this.enumType, res);
+  }
```

Comment:
We call `this.handleError` from our concrete handler with the error and the response object